code/game/objects/items/toys.dm

```diff
@@ -826,6 +824,29 @@
 	update_appearance()
 	return card_to_draw
 
+/**
+ * Sets the parent deck for this card, registering a signal to clear references on the parent deck's deletion.
+ *
+ * Arguments:
+ * * new_parent_deck - New deck to set as a parent.
+ */
+/obj/item/toy/cards/proc/set_parent_deck(obj/item/toy/cards/deck/new_parent_deck)
+	if(parentdeck)
+		UnregisterSignal(parentdeck, COMSIG_PARENT_QDELETING)
+
+	if(new_parent_deck)
+		RegisterSignal(parentdeck, COMSIG_PARENT_QDELETING, .proc/on_parent_deck_deletion)
+
+	parentdeck = new_parent_deck
```

Comment:
I'm kinda weary of using signals for this stuff because of the chance for conflicts. this seems like the perfect usecase for weakrefs even, cause we never actually need to resolve them

Reply:
We'd still need to resolve them for all the comparisons, right?

![image](https://user-images.githubusercontent.com/24975989/149226382-b8f3e0c3-afb7-44a5-bfd7-aab3badb2ef5.png)

Reply:
nah, cause we're comparing against other weakrefs most of the time, and any one object can only have one weakref
you'd need it for this check, but that could just be `card_to_add.parentdeck != WEAKREF(src)`
![image](https://user-images.githubusercontent.com/58055496/149232524-bf839334-1ad5-44f6-bd5a-a2336b8f24ab.png)